code/code.py

```diff
@@ -87,11 +84,9 @@ def code_libraries(m) -> str:
 class code_actions:
     def language():
         result = ""
-        if not forced_language:
-            file_extension = actions.win.file_ext()
-
-            if file_extension and file_extension in extension_lang_map:
-                result = extension_lang_map[file_extension]
+        file_extension = actions.win.file_ext()
```

Comment:
I feel like there's scenarios where both auto_lang and force lang could be active with this as-is. Maybe worth checking the auto_lang mode here for sanity?

Reply:
In which scenario are you thinking about?
As the code is written now you disable auto_lang when a specific language mode is forced/enabled and you enable auto_mode when forced language is cleared/disabled. So they should be mutually exclusive.

Reply:
Sorry, you're correct. Handled by the context checks.

```
mode: user.auto_lang
and code.language: javascript
```

Reply:
Yes exactly. As far as I can tell this should work.